numba/tests/test_comprehension.py

```diff
@@ -251,6 +225,166 @@ def list24(x):
             msg = "Invalid usage of == with parameters"
             self.assertIn(msg, str(raises.exception))
 
+class TestArrayComprehension(unittest.TestCase):
+
+    @tag('important')
+    def test_comp_with_array_1(self):
```

Comment:
I am looking into the this and experimenting with the use of the "refine_map" in the typeinfer to refine the array dtype just like how we did it with list.  Basic concept works but it is breaking a lot of tests.  I will update this when I make more progress.